app/src/main/java/com/xmartlabs/template/ui/mvp/BaseMvpActivity.java

```diff
@@ -0,0 +1,71 @@
+package com.xmartlabs.template.ui.mvp;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+
+import com.f2prateek.dart.Dart;
+import com.trello.rxlifecycle.components.RxActivity;
+import com.xmartlabs.template.BaseProjectApplication;
+
+/**
+ * Activity that uses a {@link MvpPresenter} to implement the MVP pattern
+ * The activities that inherit from this class will conform the V part of the said pattern
+ * @param <V> The contract that provides the public API for the presenter to invoke view related methods
+ * @param <P> The presenter that coordinates the view
+ */
+public abstract class BaseMvpActivity<V extends MvpView, P extends MvpPresenter<V>> extends RxActivity implements MvpView {
+  private P presenter;
+  protected Context getContext() {
+    return this;
+  }
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    Dart.inject(this);
+    BaseProjectApplication.getContext().inject(this);
+    presenter = createPresenter();
+    presenter.attachView(getView());
+  }
+
+  @Override
+  protected void onDestroy() {
+    super.onDestroy();
+    presenter.detachView();
+  }
+
+  /**
+   * Returns the presenter instance
+   * @return the presenter instance
+   */
+  public P getPresenter() {
```

Comment:
Why didn't you use lombok? Also I don't know if public is the best approach, may be it should be protected.